docs/reverse-proxy.md

```diff
@@ -0,0 +1,43 @@
+# Running behind a reverse proxy
+
+Running Headscale behind a reverse proxy is suitable for container-based deployments. This is especially useful on a server were port 443 is already being used for other web services.
+
+Headscale can be configured not to use TLS, leaving it to the reverse proxy to handle. Add the following configuration values to your headscale config file.
+
```

Comment:
Can you please point out that for Headscale v0.17.x+ having the reverse proxy configured to support WebSockets is mandatory? Otherwise Tailscale clients v1.30+ will not work.

WebSockets is also required when running the embedded DERP server.

In the embedded DERP case, even if you run a reverse proxy you will also have to open the specified STUN UDP port (see https://github.com/juanfont/headscale/blob/main/config-example.yaml#L85)

Reply:
I added a section saying that web socket configuration is mandatory going forward.